src/repository/db/db_repository.go

```diff
@@ -0,0 +1,260 @@
+package db
+
+import (
+	"encoding/csv"
+	"fmt"
+	"github.com/spf13/viper"
+	"golang-bootcamp-2020/domain/model"
+	_errors "golang-bootcamp-2020/utils/error"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+var (
+	isCsvFetched  = false
+	charactersMap map[string]*model.Character
+)
+
+const (
+	chId = iota
+	chName
+	chStatus
+	chSpecies
+	chType
+	chGender
+	chOriginName
+	chOriginUrl
+	chLocationName
+	chLocationUrl
+	chImage
+	chEpisodes
+
+	errorDbEmpty     = "db empty, fetch is needed"
+	errorWritingFile = "error writing csv file"
+)
+
+type dbRepository struct {
+	mapReader *io.Reader
+}
+
+type DataBaseRepository interface {
+	CreateCharactersCSV(characters []model.Character) _errors.RestError
+	GetCharacterFromId(id string) (*model.Character, _errors.RestError)
+	GetCharacters() ([]model.Character, _errors.RestError)
+	GetCharacterIdByName(name string) (string, _errors.RestError)
+}
+
+//Set the initial state of db repository
```

Comment:
By convention, all the descriptions should start with the name of the struct, interface, function, etc name
Please change all the descriptions